models/mobilenet/mobilenet_v1_impl.hpp

```diff
@@ -0,0 +1,191 @@
+    if (imageSizeString == imageSizeToString.end())
+    {
+      mlpack::Log::Fatal << "Image size can only be one of the following when"
+          " pre-trained is true: (128, 160, 192, 224)" << std::endl;
+    }
+
+    std::map<double, std::string>::iterator alphaString =
+        alphaToString.find(alpha);
+    if (alphaString == alphaToString.end())
+    {
+      mlpack::Log::Fatal << "Alpha can only be one of the following when"
+          " pre-trained is true: (0.25, 0.5, 0.75, 1.0)" << std::endl;
+    }
+
+    std::string home = getenv("HOME");
+    preTrainedPath = home + "/.cache/mlpack/models/weights/mobilenetv1/"
+        "mobilenetv1_" + alphaString->second + "_" + imageSizeString->second +
+        ".bin";
+    if (Utils::PathExists(preTrainedPath, true) == false)
+    {
+      std::cout << "mobilenetv1_" + alphaString->second + "_" +
+          imageSizeString->second + ".bin to " + preTrainedPath << std::endl;
+      Utils::DownloadFile("mobilenetv1_" + alphaString->second + "_" +
+          imageSizeString->second + ".bin", preTrainedPath, "", false, false,
+          "http://models.mlpack.org/mobilenetv1/");
+    }
+
+    LoadModel(preTrainedPath);
+    return;
+  }
+  outSize = size_t(32 * alpha);
+  mobileNet.Add(new ann::Convolution<>(inputChannel, outSize, 3, 3, 2, 2,
+      std::make_tuple(0, 1), std::make_tuple(0, 1), inputWidth, inputHeight));
+  mlpack::Log::Info << "Convolution: " << "(" << "3, " << inputWidth + 1 << ", "
+      << inputHeight + 1 << ")" << " ---> (" << outSize << ", ";
+  inputWidth = ConvOutSize(inputWidth, 3, 2, 1);
+  inputHeight = ConvOutSize(inputHeight, 3, 2, 1);
+  mlpack::Log::Info << inputWidth << ", " << inputHeight << ")" << std::endl;
+  mobileNet.Add(new ann::BatchNorm<>(outSize, 1e-3, true));
+  mlpack::Log::Info << "BatchNorm: " << "(" << outSize << ")"
+        << " ---> (" << outSize << ")" << std::endl;
+  ReLU6Layer();
+  outSize = DepthWiseConvBlock(outSize, 64, alpha, depthMultiplier);
+
+  for (const auto& blockConfig : mobileNetConfig)
+  {
+    outSize = DepthWiseConvBlock(outSize, blockConfig.first, alpha,
+        depthMultiplier, 2);
+
+    for (size_t numBlock = 1; numBlock < blockConfig.second; ++numBlock)
+    {
+      outSize = DepthWiseConvBlock(outSize, blockConfig.first, alpha,
+          depthMultiplier);
+    }
+  }
+
+  mobileNet.Add(new ann::AdaptiveMeanPooling<>(1, 1));
+  mlpack::Log::Info << "Adaptive mean pooling: (" << size_t(1024 * alpha)
+      << ", " << inputWidth << ", " << inputHeight << ") ---> ("
+      << size_t(1024 * alpha) << ", 1, 1)" << std::endl;
+
+  if (includeTop)
+  {
+    mobileNet.Add(new ann::Dropout<>(1e-3));
+    mlpack::Log::Info << "Dropout" << std::endl;
+    mobileNet.Add(new ann::Convolution<>(size_t(1024 * alpha), numClasses, 1,
+        1, 1, 1, 0, 0, 1, 1, "same"));
+    mlpack::Log::Info << "Convolution: (" << size_t(1024 * alpha)
+        << ", 1, 1) ---> (" << numClasses << " , 1, 1)" << std::endl;
```

Comment:
The above for convolution can be used here as well.

Reply:
do you think it would be fine if we could leave this as is, this would make us write some conditions inside the convblock for when to use the 1, 1 passes by us.

Reply:
We can also just use input params and init them by default with input width and height but still seems like an overkill for such a simple addition.